schnorr_fun/src/frost.rs

```diff
@@ -25,13 +26,21 @@
 //! # let secret_poly3 = frost::generate_scalar_poly(threshold, &mut rng);
 //! # let public_poly2 = frost::to_point_poly(&secret_poly2);
 //! # let public_poly3 = frost::to_point_poly(&secret_poly3);
+//! let device_index = s!(1).public();
+//! # let device_index2 = s!(2).public();
+//! # let device_index3 = s!(3).public();
```

Comment:
@LLFourn I'd be interested in your thoughts on this API.

It's extra work for the user and this macro could be foreign to people.

However we can now enforce `type PartyIndex = Scalar<Public, NonZero>;` as the indexing parameters for a lot of functions. Now since our indexes start from 1 and we get rid of a bunch of `i+1`s in the polynomial/lagrange evaluations

I like it from an instructional point of view, evaluating the polynomials at some scalar index.